fix error where users who received a no-id-professional certificate d…

chrisndodge opened the pull request:
…id not get a link on his/her dashboard to download the PDF

chrisndodge commented:
@adampalay @wedaly @clintonb 

I'm pretty sure this is the fix for https://openedx.atlassian.net/browse/ECOM-2277. I write a unit test to verify the failure first and then it passed when I made the code change.

Is there a hotfix going out on Monday? I believe I heard @doctoryes mention something about a hotfix. I just created the branch (off of 'release') as I didn't see another hotfix branch with a name of hotfix/2015-09-14

Not sure if there is a simple way to validate on stage.

I don't believe this bug was created by Solutions, so I'm wondering if someone could chip in and shepherd this the release process...

FYI, @explorerleslie @griffresch @rock345

clintonb commented on common/djangoapps/student/tests/test_certificates.py:
```diff
@@ -42,6 +42,16 @@ def test_display_verified_certificate(self, enrollment_mode):
         self._create_certificate(enrollment_mode)
         self._check_can_download_certificate()
 
+    @ddt.data('no-id-professional')
```
No need for `ddt` if you only have one variable.

clintonb commented:
@chrisndodge the hotfix in question is a security fix originating from edx-platform-secure. It has been validated and is set to go to production Monday morning. Unless there is another patch, you'll have to start a new one.

chrisndodge commented:
@clintonb addressed your comment (plus fixed pylint error). Not sure who else might want to look at this. @wedaly do you?

Also, any recommendations on how to validate in stage? How are PDF certificates normally validated in stage?

clintonb commented:
:+1:

Someone in DevOps might be able to help with validation.

adampalay commented:
@clintonb , why can't we tack this onto the security fix?

clintonb commented:
The security fix is validated and ready to go to production. The only reason it's not out now is the fact that validation was finished around 4PM on a Friday.

chrisndodge commented:
Closing, this was merged in an already existing RC
